include/hba.h

```diff
@@ -16,8 +16,55 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-struct HBA;
+#define NAME_ALL        1
+#define NAME_SAMEUSER   2
 
-struct HBA *hba_load_rules(const char *fn);
+enum RuleType {
+	RULE_LOCAL,
+	RULE_HOST,
+	RULE_HOSTSSL,
+	RULE_HOSTNOSSL,
+};
+
+struct NameSlot {
+	size_t strlen;
+	char str[];
+};
+struct HBAName {
+	unsigned int flags;
+	struct StrSet *name_set;
+};
+
+struct HBARule {
+	struct List node;
+	enum RuleType rule_type;
+	int rule_method;
+	int rule_af;
+	uint8_t rule_addr[16];
+	uint8_t rule_mask[16];
+	struct HBAName db_name;
+	struct HBAName user_name;
+	struct IDENTMap *identmap;
+};
+
+struct HBA {
+	struct List rules;
+};
+
+struct IDENTMap {
```

Comment:
Let's use CamelCase for the struct name. I realize took the naming scheme from HBA, but that is an acronym (and even then I don't agree with this type of casing personally).
```suggestion
struct IdentMap {
```